sklearn/svm/_classes.py

```diff
@@ -1,14 +1,35 @@
 from numbers import Integral, Real
+import warnings
 
 import numpy as np
 
-from ._base import _fit_liblinear, BaseSVC, BaseLibSVM
+from ._base import _fit_liblinear, _get_liblinear_solver_type, BaseSVC, BaseLibSVM
 from ..base import BaseEstimator, RegressorMixin, OutlierMixin
 from ..linear_model._base import LinearClassifierMixin, SparseCoefMixin, LinearModel
 from ..utils import deprecated
 from ..utils.validation import _num_samples
 from ..utils.multiclass import check_classification_targets
-from ..utils._param_validation import Interval, StrOptions
+from ..utils._param_validation import Interval, StrOptions, Hidden
+
+
+def _choose_dual_automatically(loss, penalty, multi_class, X):
+    """Choose dual parameter value wrt to loss, penalty and shape of the data"""
+    if X.shape[0] < X.shape[1]:
+        try:
+            # check if the combination of parameters supports
+            # dual formulation
+            _get_liblinear_solver_type(multi_class, penalty, loss, True)
+            return True
+        except ValueError:
+            return False
```

Comment:
Apparently, we don't have a test for this case. It could be nice to hand-craft a case that this one happen. Otherwise, it means that we don't need the `try ... except` here.